### PR TITLE
Navigation branch

### PR DESCRIPTION
**Pull request by mlynch5187**
This branch deals encompasses all user stories 2-9. These user stories deal with navigation. 

- There are now 3 roles of users: user, merchant, and admin
- Each user role has specific pages available to it, and others they cannot see
- Each user role also has its own dashboard or profile page
- There is now a navigation bar at the top of each page
- All test passing at 99.93% coverage

### CODE REVIEW

File: app/controllers/cart_controller.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 class CartController < ApplicationController
+
+  before_action :require_not_admin
+
+  def require_not_admin
+    render file: "/public/404" if current_admin?
+  end
```
Comment (reid-andrew) [inline app/controllers/cart_controller.rb:7]: Blocking admin from navigating to cart pages.

File: app/controllers/merchant/dashboard_controller.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  before_action :require_merchant
+
+  def require_merchant
+    render file: "/public/404" unless current_merchant?
+  end
+
+  def index
+
+  end
```
Comment (reid-andrew) [inline app/controllers/merchant/dashboard_controller.rb]: Potential for small refactor here to get this on one line.

Comment (reid-andrew) [reply app/controllers/merchant/dashboard_controller.rb]: We've done one-line approach in other controllers.

File: app/controllers/admin/dashboard_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Admin::DashboardController < ApplicationController
+
+  before_action :require_admin
+
+  def require_admin
+    render file: "/public/404" unless current_admin?
+  end
+
+  def index
+  end
```
Comment (javier-aguilar) [inline app/controllers/admin/dashboard_controller.rb]: To stick to convention, we could do
` def index; end`

File: app/controllers/merchant/dashboard_controller.rb (hunk 2 above)
Comment (mlynch5187) [reply app/controllers/merchant/dashboard_controller.rb]: will do!

File: app/controllers/admin/dashboard_controller.rb (hunk 3 above)
Comment (mlynch5187) [reply app/controllers/admin/dashboard_controller.rb]: We're on it!

**Review by javier-aguilar — APPROVED** [review]

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -10,10 +10,26 @@
...
+         <%= link_to "Register", "/register" %>
+       <% end %>
+       <% if current_merchant? %>
+        <%= link_to "Merchant Dashboard" %>
+      <% end %>
+      <% if current_admin? %>
+        <%= link_to "Admin Dashboard" %>
+        <%= link_to "All Users" %>
+     <% end %>
+     </nav>
```
Comment (aperezsantos) [inline app/views/layouts/application.html.erb:32]: Could benefit from adding spacing?

**Review by reid-andrew — APPROVED** [review]

**Review by aperezsantos — APPROVED** [review]
This looks great. There may be some spacing improvements.

**Comment by mlynch5187** [comment]
I made some suggested changes to the empty def/end methods to make them one-liners

File: app/controllers/admin/dashboard_controller.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Admin::DashboardController < ApplicationController
+
+  before_action :require_admin
+
+  def require_admin
+    render file: "/public/404" unless current_admin?
+  end
+
+  def index; end
```
Comment (mlynch5187) [inline app/controllers/admin/dashboard_controller.rb:9]: Updated per Andy's request

**Pull request merged**